.github/workflows/ci.yml

```
@@ -0,0 +1,41 @@
+name: CI/CD pipeline for Flask App
+
+on:
+  push:
+    branches: [ main ]
+    pull_request:
```

actionlint (1.7.4)

unexpected key "pull_request" for "push" section. expected one of "branches", "branches-ignore", "paths", "paths-ignore", "tags", "tags-ignore", "types", "workflows"

(syntax-check)